kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -186,6 +199,30 @@ private void updateMetadataWithICTIfRequired(Engine engine, CommitInfo attemptCo
         );
     }
 
+    private Optional<Long> getUpdatedInCommitTimestampAfterConflict(
+            long winningCommitTimestamp, Optional<Long> attemptInCommitTimestamp) {
+        if (attemptInCommitTimestamp.isPresent()) {
+            long updatedInCommitTimestamp = Math.max(
+                    attemptInCommitTimestamp.get(), winningCommitTimestamp + 1);
+            return Optional.of(updatedInCommitTimestamp);
+        }
+        return attemptInCommitTimestamp;
+    }
+
+    private Optional<Metadata> getMetadataWithUpdatedICTEnablementInfo(
+            Engine engine,
+            Long updatedInCommitTimestamp,
+            Metadata attemptMetadata,
+            Long lastWinningVersion) {
+        long nextAvailableVersion = lastWinningVersion + 1L;
+        return InCommitTimestampUtils.getUpdatedMetadataWithICTEnablementInfo(
+                engine,
+                updatedInCommitTimestamp,
+                readSnapshot,
+                attemptMetadata,
+                nextAvailableVersion);
+    }
```

Comment:
This method isn't adding any value. Use call the `InCommitTimestampUtils.getUpdatedMetadataWithICTEnablementInfo` directly.